### PR TITLE
Cloud Monitoring: Out-of-the-box dashboards

### PR DESCRIPTION
**Pull request by papagian**
**What this PR does / why we need it**:
Adds out-of-the-box dashboards for the Cloud Monitoring datasource

These dashboards have been derived from the respective Monitoring ones (also available in [CGP console](https://console.cloud.google.com/monitoring/dashboards)):

- [BigQuery Query Monitoring](https://github.com/GoogleCloudPlatform/monitoring-dashboard-samples/blob/master/dashboards/bigdata/bigquery-monitoring.json)
- [Cloud SQL Monitoring](https://github.com/GoogleCloudPlatform/monitoring-dashboard-samples/blob/master/dashboards/storage/cloudsql-monitoring.json)
- [GCE VM Instance Monitoring](https://github.com/GoogleCloudPlatform/monitoring-dashboard-samples/blob/master/dashboards/compute/gce-vm-instance-monitoring.json)
- [GKE Cluster Monitoring](https://github.com/GoogleCloudPlatform/monitoring-dashboard-samples/blob/master/dashboards/compute/gke-cluster-monitoring.json)
- [HTTP_S Load Balancer Monitoring](https://github.com/GoogleCloudPlatform/monitoring-dashboard-samples/blob/master/dashboards/networking/https-loadbalancer-monitoring.json)

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

- [X] A complementary PR adding the new image for the documentation will follow up.

### CODE REVIEW

File: docs/sources/features/datasources/cloudmonitoring.md
Comment (wbrowne) [inline docs/sources/features/datasources/cloudmonitoring.md:343]: Should `data source` be `datasource` in this case? 🤔 Same for below on line 351

File: public/app/plugins/datasource/cloud-monitoring/plugin.json
Comment (wbrowne) [inline public/app/plugins/datasource/cloud-monitoring/plugin.json:11]: I'm wondering if it's safer to have filenames without spaces?

File: docs/sources/features/datasources/cloudmonitoring.md
Comment (oddlittlebird) [reply docs/sources/features/datasources/cloudmonitoring.md:343]: Yes, per the [Word usage section of the Grafana documentation style guide](https://github.com/grafana/grafana/blob/master/contribute/style-guides/documentation-style-guide.md#setup-set-up).

File: docs/sources/features/datasources/cloudmonitoring.md
Comment (oddlittlebird) [inline docs/sources/features/datasources/cloudmonitoring.md:341]: Please do not merge this until the 7.3 beta. @daniellee , FYI.

File: docs/sources/features/datasources/cloudmonitoring.md
Comment (oddlittlebird) [inline docs/sources/features/datasources/cloudmonitoring.md:353]:
```suggestion
The dashboards have a template variable which is populated with the projects accessible by the configured service account every time the dashboard is loaded. After the dashboard is loaded, you can select the project you prefer from the drop-down list.
```

**Review by oddlittlebird — DISMISSED** [review]

File: public/app/plugins/datasource/cloud-monitoring/plugin.json
Comment (oddlittlebird) [reply public/app/plugins/datasource/cloud-monitoring/plugin.json:11]: @marcusolsson, are there plugin-specific recommendations? If not, should there be?

Comment (marefr) [reply public/app/plugins/datasource/cloud-monitoring/plugin.json:11]: Without spaces and lowercase is probably the safest

File: docs/sources/features/datasources/cloudmonitoring.md
Comment (marefr) [reply docs/sources/features/datasources/cloudmonitoring.md:341]: Why? Have we changed this approach? Not optimal having PR's ready, but not merged for some time and then they're going to be merged for the beta. We want to merge as fast possible to make sure it works as expected in master. Guess the approach would be to open separate PR's for documentation then?

File: docs/sources/features/datasources/cloudmonitoring.md
Comment (marcusolsson) [reply docs/sources/features/datasources/cloudmonitoring.md:343]: From the [style guide](https://github.com/grafana/grafana/blob/master/contribute/style-guides/documentation-style-guide.md#data-source):

> Two words, not one

File: public/app/plugins/datasource/cloud-monitoring/plugin.json
Comment (papagian) [reply public/app/plugins/datasource/cloud-monitoring/plugin.json:11]: thank you all for your comments; I will change them to lowercase without spaces.
I'm thinking also that `/` separator can also be an issue in Windows but I don't know how to change it to work for both. This must be [where](https://github.com/grafana/grafana/blob/master/pkg/plugins/dashboards.go#L99) it is consumed. Any ideas?

File: docs/sources/features/datasources/cloudmonitoring.md
Comment (oddlittlebird) [reply docs/sources/features/datasources/cloudmonitoring.md:341]: Literally having a meeting today to discuss it, yes. And I'm just talking about docs, not the feature.

Which is not ideal, I know. I would like the docs to stay in lockstep with the code, just as this PR has it, but right now we are trying to figure out how to keep the future-state stuff out of the current docs as much as possible (currently 7.2 docs) without driving us all insane.

File: public/app/plugins/datasource/cloud-monitoring/plugin.json
Comment (papagian) [reply public/app/plugins/datasource/cloud-monitoring/plugin.json]: I have tested importing dashboards in windows (from the [cloudwatch datasource](https://github.com/grafana/grafana/blob/b892aa2eb73f64f49f217f6f7c74f37cd9e73959/public/app/plugins/datasource/cloudwatch/plugin.json#L11)) and it works so the `/` separator is not a problem (as I expected).

File: docs/sources/features/datasources/cloudmonitoring.md
Comment (achatterjee-grafana) [inline docs/sources/features/datasources/cloudmonitoring.md:345]: Remove comma after bullets. No period after the last bullet.

Comment (achatterjee-grafana) [reply docs/sources/features/datasources/cloudmonitoring.md:345]: Also, the topic should have lazy numbering for all procedure sections. Need correction.

Comment (papagian) [reply docs/sources/features/datasources/cloudmonitoring.md:345]: what do you mean by "lazy numbering". Can you be more specific or point me to an example? thanks!

Comment (achatterjee-grafana) [reply docs/sources/features/datasources/cloudmonitoring.md:345]: All numbering should be 1, and not 1,2,3, ... they will render correctly in your preview.

Here is an explanation:

Use lazy numbering for long lists
Markdown is smart enough to let the resulting HTML render your numbered lists correctly. For longer lists that may change, especially long nested lists, use “lazy” numbering:

1.  Foo.
1.  Bar.
    1.  Foofoo.
    1.  Barbar.
1.  Baz.

File: docs/sources/features/datasources/cloudmonitoring.md
Comment (torkelo) [reply docs/sources/features/datasources/cloudmonitoring.md:341]: Until we have solved https://github.com/grafana/grafana/issues/27384 we are stuck with version notices (which can be good even post that)

Comment (marefr) [reply docs/sources/features/datasources/cloudmonitoring.md:341]: @papagian which means we can merge this before 7.3 beta leaving the version notes intact.

File: public/app/plugins/datasource/cloud-monitoring/dashboards/loadBalancer.json
Comment (marefr) [inline public/app/plugins/datasource/cloud-monitoring/dashboards/loadBalancer.json:1129]: Compared to the other title which are very GCP centrix, this one is more general and may clash with users existing dashboards. Not sure how big a problem that is though.

**Review by marefr — APPROVED** [review]
LGTM, but left a comment/questions.

Have a general question as well. Has there been any discussions about using a data source variable to support multiple data source instances? Now you can only import the dashboard and use for that specific data source you're importing from. Besides selecting a project, being able to select which Cloud Monitoring data source in a drop down before project could be useful, I think that's how CloudWatch dashboards have been implemented.

**Comment by papagian** [comment]
> LGTM, but left a comment/questions.
> 
> Have a general question as well. Has there been any discussions about using a data source variable to support multiple data source instances? Now you can only import the dashboard and use for that specific data source you're importing from. Besides selecting a project, being able to select which Cloud Monitoring data source in a drop down before project could be useful, I think that's how CloudWatch dashboards have been implemented.

Initially I had such a datasource template variable for selecting datasource as well but I change it
because in order to import such a dashboard you need to select first a cloud monitoring datasource
(and then navigating to the Dashboards tab)
it made more sense to me to associate the dashboard to the specific datasource.
However, I'm open to change it this it we decide that is more useful.

File: public/app/plugins/datasource/cloud-monitoring/dashboards/loadBalancer.json
Comment (papagian) [reply public/app/plugins/datasource/cloud-monitoring/dashboards/loadBalancer.json:1129]: I just used the original [title](https://github.com/GoogleCloudPlatform/monitoring-dashboard-samples/blob/master/dashboards/networking/https-loadbalancer-monitoring.json#L2).
What about `HTTP/S GCP Load Balancer Monitoring`?

**Comment by marefr** [comment]
> Initially I had such a datasource template variable for selecting datasource as well but I change it
because in order to import such a dashboard you need to select first a cloud monitoring datasource
(and then navigating to the Dashboards tab)

Not sure I follow. Given you have one or several cloud monitoring data sources configured you can go into one of these, dashboards tab and import the dashboards and it will work for all your cloud monitoring data sources.

> it made more sense to me to associate the dashboard to the specific datasource.
> However, I'm open to change it we decide that is more useful.

I guess the limitation with this approach is that you can only use the dashboards with one data source at a time. Importing a dashboard for data source A it will work for data source A. Importing it again from data source B it will stop working for data source A and work for data source B.

**Comment by papagian** [comment]
> > Initially I had such a datasource template variable for selecting datasource as well but I change it
> > because in order to import such a dashboard you need to select first a cloud monitoring datasource
> > (and then navigating to the Dashboards tab)
> 
> Not sure I follow. Given you have one or several cloud monitoring data sources configured you can go into one of these, dashboards tab and import the dashboards and it will work for all your cloud monitoring data sources.
> 
> > it made more sense to me to associate the dashboard to the specific datasource.
> > However, I'm open to change it we decide that is more useful.
> 
> I guess the limitation with this approach is that you can only use the dashboards with one data source at a time. Importing a dashboard for data source A it will work for data source A. Importing it again from data source B it will stop working for data source A and work for data source B.

@marefr
I updated the dashboards and I've added the datasource variable to support multiple data source instances
do you want to have a fresh look?

**Review by marefr — APPROVED** [review]
LGTM

**Comment by papagian** [comment]
@oddlittlebird
doc modifications look good to you now?
If yes, can you approve them? thanks!

**Review by achatterjee-grafana — APPROVED** [review]
Approving, assuming the changes Diana(@oddlittlebird) and Marcus asked for, were made. Thank you!

**Pull request merged**
